tileview/src/main/java/com/qozix/tileview/tiles/TileCanvasViewGroup.java

```diff
@@ -262,12 +268,18 @@ void generateTileBitmap( Tile tile ) {
     tile.generateBitmap( getContext(), getBitmapProvider() );
   }
 
-  void addTileToCurrentTileCanvasView( Tile tile ) {
-    tile.setTransitionsEnabled( mTransitionsEnabled );
-    tile.setTransitionDuration( mTransitionDuration );
-    tile.stampTime();
-    mTilesAlreadyRendered.add( tile );
-    mCurrentTileCanvasView.addTile( tile );
+  void addTileToCurrentTileCanvasView( final Tile tile ) {
+    mTileRenderHandler.post( new Runnable() {
+      @Override
+      public void run() {
+        tile.setTransitionsEnabled( mTransitionsEnabled );
+        tile.setTransitionDuration( mTransitionDuration );
+        tile.stampTime();
+        mTilesAlreadyRendered.add( tile );
+        mCurrentTileCanvasView.addTile( tile );
```

Comment:
Created a `PrepareTileForRenderRunnable` class and a method `prepareTileForRender` on `TileViewGroup`for that. I believe solves both "issues" regarding the same peace of code